transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala

```diff
@@ -281,21 +287,31 @@ object HcaPipelineBuilder extends PipelineBuilder[Args] {
     isFileMetadata: Boolean = false
   ): ClosedTap[String] = {
     // get the readable files for the given input path
-    val readableFiles = getReadableFiles(
-      s"$inputPrefix/metadata/${entityType}/**.json",
-      context
-    )
-    // then convert json to msg and get the filename
-    val processedData = jsonToFilenameAndMsg(entityType)(readableFiles)
-      .withName(s"Pre-process ${entityType} metadata")
-      .map {
-        case (filename, metadata) =>
-          if (isFileMetadata) transformFileMetadata(entityType, filename, metadata)
-          else transformMetadata(entityType, filename, metadata)
-      }
+    val metadataFiles = getReadableFiles(s"$inputPrefix/metadata/${entityType}/**.json", context)
+    val metadataFilenameAndMsg = jsonToFilenameAndMsg(entityType)(metadataFiles)
+
+    // for file metadata
+    val readyMetadata = if (isFileMetadata) {
```

Comment:
Would be nice if we could unit-test from here down (`ReadableFile` is declared `final` so I don't think we can mock it to include `jsonToFilenameAndMsg` in a unit-test).

Could you think through what it might look like to shift this business logic into a separate class / method? Ideally it'd be unit-testable independently from the implementations of `transformMetadata` and `transformFileMetadata` (we'd need some interfaces in front of those which we could mock).